src/main/java/org/codehaus/plexus/util/ReflectionUtils.java

```diff
@@ -218,6 +219,14 @@ private static void gatherVariablesAndValuesIncludingSuperclasses( Object object
 
         Class<?> clazz = object.getClass();
 
+        if ( Integer.parseInt( System.getProperty( "java.specification.version" ).split( "\\." )[0] ) >= 11
```

Comment:
Don't we have anything better in place here?

Reply:
> Don't we have anything better in place here?

```java
Float.parseFloat( System.getProperty( "java.specification.version" ) )
```

But `Runtime.Version` ([doc](https://docs.oracle.com/javase/9/docs/api/java/lang/Runtime.Version.html)) cannot be used, exist since Java 9 (`plexus-utils` always compatible with Java 7). And without adding some third-part tooling as dependencies, I don't see another way 😢.

I will update commit with *Float* instead *Integer*.

Reply:
> I will update commit with Float instead Integer.

Done

Reply:
@michael-o : Or did you want a better implementation around `System.getProperty( "java.specification.version" )` if value is *null/blanck/notParsable* ?